### ISSUE TITLE
💅 Incorrect Linting Suggestion for useLiteralKeys with __proto__ as Computed Property

### ISSUE THREAD

**Issue by vickaita** [biomejs/biome#2430]
### Environment information

```bash
CLI:
  Version:                      1.6.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.1.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        correctness/noUnusedImports = "error"
                                correctness/useHookAtTopLevel = "error"
                                style/noInferrableTypes = "off"
                                style/useForOf = "error"
                                style/useImportType = "off"
                                suspicious/noArrayIndexKey = "off"

Workspace:
  Open Documents:               0
```

### Rule name

lint/complexity/useLiteralKeys

### Playground link

https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAbwBiAGoAIAA9ACAAewBbACIAXwBfAHAAcgBvAHQAbwBfAF8AIgBdADoAIAB0AHIAdQBlAH0AOwA%3D

### Expected result

The useLiteralKeys rule suggests that object literals with computed properties such as `{["__proto__"]: true}` can be simplified to `{__proto__: true}`. However, this suggestion is incorrect due to the different ways JavaScript handles these syntaxes. In JavaScript, `{["__proto__"]: true}` does not modify the object's prototype, whereas `{__proto__: true}` does, potentially leading to unexpected behavior.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Thank you JavaScript 😞

**Comment by Conaclos** [comment]
`__proto__` is a [non-standard and deprecated](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/proto) feature.
However, I accept that we could ignore this because this has security implication.

By the way, why do you want to use a such name as property?

**Comment by vickaita** [comment]
@Conaclos I was implementing a `clone` function and this came up as an edge case. I wouldn't normally use `__proto__` as a property name, but I needed it in an unit test for the edge case.
